packages/edit-site/src/components/sidebar-edit-mode/template-card/index.js

```diff
@@ -14,10 +19,37 @@ import { store as editSiteStore } from '../../../store';
 import TemplateActions from './template-actions';
 import TemplateAreas from './template-areas';
 
+const MaybeBlockPatterns = ( { template, blockPatterns, title } ) => {
+	if ( template.type !== 'wp_template_part' || blockPatterns?.length === 0 ) {
+		return null;
+	}
+
+	return (
+		<>
+			<p>
+				{ sprintf(
+					// translators: %s: Type of template part (i.e. header, footer, etc.)
+					__(
+						'Choose a pre-designed pattern to switch up the look and feel of your %s '
+					),
+					title.toLowerCase()
+				) }
+			</p>
+			<BlockPatternsList
+				blockPatterns={ blockPatterns }
+				shownPatterns={ blockPatterns }
+				onClickPattern={ ( pattern ) => {
+					console.log( 'pattern clicked ', pattern );
```

Comment:
TODO: need to implement actually replacing the template part with the pattern's content.